src/group_theory/torsion.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2022 Julian Berman. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Berman
+-/
+
+import group_theory.exponent
+import group_theory.order_of_element
+import group_theory.quotient_group
+
+/-!
+# Torsion groups
+
+This file defines torsion groups, i.e. groups where all elements have finite order.
+
+## Main definitions
+
+* `monoid.is_torsion` is a predicate asserting a monoid `G` is a torsion monoid, i.e. that all
+  elements are of finite order. Torsion groups are also known as periodic groups.
+* `add_monoid.is_torsion` the additive version of `monoid.is_torsion`.
+
+## Main results
+
+-/
+
+universe u
+
+variable {G : Type u}
+
+open_locale classical
+
+namespace monoid
+
+variables (G) [monoid G]
+
+/--A predicate on a monoid saying that all elements are of finite order.-/
+@[to_additive "A predicate on an additive monoid saying that all elements are of finite order."]
+def is_torsion := ∀ g : G, is_of_fin_order g
+
+end monoid
+
+open monoid
+
+variables [group G] {N : subgroup G}
+
+/--Subgroups of torsion groups are torsion groups. -/
+@[to_additive "Subgroups of additive torsion groups are additive torsion groups."]
+lemma subgroup.is_torsion {tG : is_torsion G} (H : subgroup G) : is_torsion H := begin
+  intro g,
+  obtain ⟨n, ⟨npos, hn⟩⟩ := tG g,
+  refine ⟨n, npos, subtype.coe_injective _⟩,
+  rw is_periodic_pt_mul_iff_pow_eq_one ↑g at hn,
```

Comment:
For me it would be more natural to work entirely in the group API. Would it be much harder to use `is_of_fin_order_iff_pow_eq_one` instead?

Reply:
Oh! This definitely looks easier, somehow I missed it. Thank you! Will give those changes a shot.